### ISSUE TITLE
Remove empty anyOf

### ISSUE THREAD

**Issue by qraynaud** [StefanTerdell/zod-to-json-schema#19]
I have the following schema to add support for mongodb `ObjectId` type:

```ts
export const objectId = z.union([
  z.instanceof(ObjectId),
  z
    .string()
    .regex(/^[a-f0-9]{24}$/, { message: "This string does not look like an ObjectId()" })
    .transform((s) => new ObjectId(s))
]);
```

This translates to the following schema:
```js
{
  anyOf: [ {}, { type: 'string', pattern: '^[a-f0-9]{24}$' } ]
}
```

The issue is that `{}` allows for any input. I think it would be more appropriate to imply that `z.instanceOf()` cannot be translated into a JSON Schema and therefore should never be able to match anything somehow.

Does that make any sense?

**Comment by StefanTerdell** [comment]
Yup, makes sense. z.instanceOf should return undefined on parse and this problem will be solved.

**Comment by qraynaud** [comment]
I wanted to tackle this but I came across a small hiccup. After reading zod codebase, I think it will be hard to implement properly.

When calling `z.instanceof()`, zod creates a `ZodAny` type with a custom check function on it. Thus there is no good way to know a node is an instanceof one. It might need a small zod rewrite. Do you agree?

**Comment by StefanTerdell** [comment]
That seems reasonable, yeah. Lets keep this open for now, but looks like logging an issue / PR-ing Zod will be the first step.

**Comment by StefanTerdell** [comment]
@qraynaud Hey, I just released 3.18.1 which adds a `strictUnion` option. Let me know if this solves your problem. If so I think we can close this

**Comment by qraynaud** [comment]
Hey! This seems like a reasonable option at lest for the time being yes.
